Source/deviceinfo/DeviceInfo.cpp

```diff
@@ -26,7 +26,26 @@
 using namespace Thunder;
 namespace {
 
+deviceinfo_status_t DeviceInfoStatus(uint32_t status)
+{
+    deviceinfo_status_t deviceInfoStatus;
+    switch (status) {
+    case Core::ERROR_NONE:
+        deviceInfoStatus = deviceinfo_status::DEVICEINFO_OK;
+        break;
+    case Core::ERROR_UNAVAILABLE:
+        deviceInfoStatus = deviceinfo_status::DEVICEINFO_ERROR_UNAVAILABLE;
+        break;
+    case Core::ERROR_INVALID_INPUT_LENGTH:
+        deviceInfoStatus = deviceinfo_status::DEVICEINFO_ERROR_INVALID_INPUT_LENGTH;
+        break;
+    default:
+        deviceInfoStatus = deviceinfo_status::DEVICEINFO_ERROR_GENERAL;
```

Comment:
perhaps an assert? in case we're getting an unforeseen error message

Reply:
done